test/integration/consumer_group_test.go

```diff
@@ -86,4 +88,35 @@ func TestConsumerGroup(t *testing.T) {
 			return false
 		}
 	}, ingressWait, waitTick)
+
+	t.Logf("validating that consumer group: %q has the status subresource set properly", consumerGroupName)
+	require.Eventually(t, func() bool {
+		cg, err = c.ConfigurationV1beta1().KongConsumerGroups(ns.Name).Get(ctx, consumerGroupName, metav1.GetOptions{})
+		if err != nil {
+			t.Logf("WARNING: error while getting consumer group: %s", err)
+			return false
+		}
+		if len(cg.Status.Conditions) != 1 {
+			t.Logf("WARNING: expected 1 status condition, got: %d", len(cg.Status.Conditions))
+			return false
+		}
+		status := cg.Status.Conditions[0]
+		if status.Message != utils.ProgrammedConditionTrueMessage {
+			t.Logf("WARNING: expected message: %q, got: %q", utils.ProgrammedConditionTrueMessage, status.Message)
+			return false
+		}
+		if status.Reason != string(kongv1.ReasonProgrammed) {
+			t.Logf("WARNING: expected reason: %q, got: %q", kongv1.ReasonProgrammed, status.Reason)
+			return false
+		}
+		if status.Status != metav1.ConditionTrue {
+			t.Logf("WARNING: expected status: %q, got: %q", metav1.ConditionTrue, status.Status)
+			return false
+		}
+		if status.Type != string(kongv1.ConditionProgrammed) {
+			t.Logf("WARNING: expected type: %q, got: %q", kongv1.ConditionProgrammed, status.Type)
+			return false
+		}
+		return true
+	}, ingressWait, waitTick)
```

Comment:
Because we have an envtest covering this already, I'm on the fence about whether we need to duplicate this in integration tests as well. 🤔

Reply:
Yup, I was wondering too. I added it because it's pretty cheap to have and easy to add (especially firstly when `EventuallyWithT()` is used). I thought about testing it when the consumer, plugin, and consumer group will be configured. But as well I can remove it as, as you said it's already in envtests. Do you have a strong preference?

Reply:
Well, in my opinion, it's better to keep this kind of assertion in envtests as they're not dependent on the integration with Kong. envtest already covers the `Programmed` condition E2E. Adding the assertion to the integration test in my feeling is unnecessary duplication of this. 

I'd be interested in @pmalek opinion on that. :)

Reply:
👍 I don't see a reason either to test it if it's already covered with `envtest` based test.

The only point I'd like to make it so be sure we cover the corner cases w.r.t API differences between dbless and db backed in Admin API mocks (and ideally mention that explicitly). That was a surprise to me when I saw a special case for db backed and dbless recently in one of the PRs for Consumer Groups

Reply:
I got rid of it